mars/dataframe/core.py

```diff
@@ -1028,6 +1028,14 @@ def empty(self):
             raise ValueError('Tileable object must be executed first')
         return shape == (0,)
 
+    @property
+    def equals(self, other):
+        if not (isinstance(other, type(self)) or isinstance(self, type(other))):
+            return False
+        from .datasource.from_tensor import series_from_tensor
+        other = series_from_tensor(other)
+        return self._mgr.equals(other._mgr)
```

Comment:
There is no `_mgr` properties for Mars DataFrame, you need other way to implement this function.